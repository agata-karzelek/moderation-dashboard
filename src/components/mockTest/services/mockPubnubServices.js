import { mockAccounts } from "../mockAccounts";
import { mockPermissions } from "../mockPermissions";
import { mockPubNubApplications } from "../mockPubnubAccounts";

const commonkeyValuePairs = {
  status: 200,
  statusText: "OK",
  headers: {
    "cache-control": "no-cache, private",
    "content-type": "application/json",
    expires: "Thu, 29 Jul 2021 15:37:05 GMT",
    pragma: "private",
  },
  config: {
    url: "/apps?owner_id=572238",
    method: "get",
    headers: {
      Accept: "application/json, text/plain, */*",
      "X-Session-Token": "rAL18qRqj8rhtxkJwzQ2dmRi5UrhH-vFbqUaLknxshs",
    },
    baseURL: "https://admin.pubnub.com/api",
    transformRequest: [null],
    transformResponse: [null],
    timeout: 0,
    xsrfCookieName: "XSRF-TOKEN",
    xsrfHeaderName: "X-XSRF-TOKEN",
    maxContentLength: -1,
    maxBodyLength: -1,
  },
  request: {},
};

export const mockAllAccounts = {
  data: {
    result: {
      accounts: mockAccounts,
      permissions: mockPermissions,
    },
  },
  ...commonkeyValuePairs,
};

export const mockAllApps = {
  data: mockPubNubApplications,
  ...commonkeyValuePairs,
};

export default null;
