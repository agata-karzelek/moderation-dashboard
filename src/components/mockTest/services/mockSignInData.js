import { mockPermissions } from "../mockPermissions";

export const mockSignInData = {
  data: {
    result: {
      created: 1627276598,
      expires: 1627634943,
      modified: 1627548543,
      role: "user",
      status: 1,
      storage: null,
      token: "dGgB0-lLGG5GmjFgtr6t2nBFqgM05Nijq_YzzYdxVUk",
      user: {
        created: 1619760527,
        deleted: 0,
        email: "sourcefusepubnub@gmail.com",
        id: 572287,
        login_attempts: 0,
      },
      user_id: 572287,
      user_roles: {
        is_internal_admin: false,
        is_internal_operator: false,
        is_internal_viewer: false,
        roles: mockPermissions,
      },
    },
  },
  status: 200,
  statusText: "OK",
};

export default null;
