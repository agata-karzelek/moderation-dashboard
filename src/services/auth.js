import axios from "../utils/axios";

export async function signIn(credentials) {
  const signInResponse = await axios.post("/me", credentials);
  if (signInResponse.status === 200) {
    return signInResponse.data;
  }
  throw new Error(signInResponse.data.error || "Something went wrong");
}
