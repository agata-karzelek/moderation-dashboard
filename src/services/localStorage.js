import cookie from "js-cookie";

export const setCookie = (key, value) => {
  cookie.set(key, value, {
    expires: 1,
  });
};

export const removeCookie = (key) => {
  cookie.remove(key, {
    expires: 1,
  });
};

export const getCookie = (key) => {
  return cookie.get(key);
};

export const setLocalStorage = (key, value) => {
  localStorage.setItem(key, JSON.stringify(value));
};

export const removeLocalStorage = (key) => {
  localStorage.removeItem(key);
};

export const authenticate = (data, next) => {
  setCookie("token", data.token);
  setLocalStorage("user", data.user);
  next();
};

export const signout = (next) => {
  localStorage.clear();
  next();
};

export const isAuth = () => {
  const cookieChecked = getCookie("token");
  if (cookieChecked) {
    if (localStorage.getItem("user")) {
      return JSON.parse(localStorage.getItem("user"));
    } else {
      return false;
    }
  }
};
