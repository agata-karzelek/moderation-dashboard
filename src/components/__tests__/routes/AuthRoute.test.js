import React from "react";
import { shallow } from "enzyme";
import AuthRoute from "../../../routes/AuthRoute";
import { isAuth } from "../../../services/localStorage";
import { mockUser } from "../../mockTest/services/mockServices";

jest.mock("../../../services/localStorage", () => {
  return {
    isAuth: jest.fn(),
  };
});

describe("Test Cases for Auth Route", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<AuthRoute />);
  });

  afterAll(() => {
    jest.resetAllMocks();
  });

  test("Snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render router", () => {
    expect(wrapper.find("Route")).toHaveLength(1);
  });

  test("check authentication", async () => {
    isAuth.mockImplementation(() => mockUser);
    await isAuth();
    expect(isAuth).toHaveBeenCalledTimes(1);
  });
});
