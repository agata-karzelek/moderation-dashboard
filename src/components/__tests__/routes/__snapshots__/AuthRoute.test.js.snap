// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Cases for Auth Route Snapshot 1`] = `
<Route
  render={[Function]}
/>
`;
